### PR TITLE
deps: update proto-google-cloud-alloydb to v1

### LINKED ISSUE #113: Update dependency proto-google-cloud-alloydb-v1beta to v1
Update dependency `proto-google-cloud-alloydb-v1beta` to v1.

Right now the Connector-related APIs aren't in v1, but will be soon-ish. When that happens, we'll need to upgrade `proto-google-cloud-alloydb-v1beta` to v1.

### LINKED ISSUE #203: Send public key instead of CSR in the generateClientCertificate API call
Update [DefaultConnectionInfoRepository.java](https://github.com/GoogleCloudPlatform/alloydb-java-connector/blob/main/alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/DefaultConnectionInfoRepository.java#L110) to send public key instead of CSR.

I tried it today (e.g. using `setPublicKey(keyPair.getPublic().toString())`) and I got the error:

```
java.util.concurrent.ExecutionException: com.google.api.gax.rpc.InvalidArgumentException: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: 
The CreateCertificateRequest proto is invalid:\n * certificate.config.public_key: Field `key` must be explicitly specified.
```

We should wait until `GenerateClientCertificateRequest` is fixed.

### PR DESCRIPTION
**Pull request by ttosta-google**
Fixes #113

### CODE REVIEW

**Comment by enocom** [comment]
Not sure what this means:

```
java.lang.RuntimeException: com.google.api.gax.rpc.ApiException: API interaction failed
```

File: alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/DefaultConnectionInfoRepository.java [hunk 1]
```diff
@@ -62,15 +53,15 @@ class DefaultConnectionInfoRepository implements ConnectionInfoRepository {
...
     Future<GenerateClientCertificateResponse> clientCertificateResponseFuture =
         executor.submit(() -> getGenerateClientCertificateResponse(instanceName, keyPair));
 
-    com.google.cloud.alloydb.v1beta.ConnectionInfo info = infoFuture.get();
+    com.google.cloud.alloydb.v1.ConnectionInfo info = infoFuture.get();
 
     GenerateClientCertificateResponse certificateResponse = clientCertificateResponseFuture.get();
-    ByteString pemCertificateBytes = certificateResponse.getPemCertificateBytes();
+    ByteString pemCertificateBytes = certificateResponse.getPemCertificateChainBytes(0);
     X509Certificate clientCertificate = parseCertificate(pemCertificateBytes);
```
Comment (enocom) [inline alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/DefaultConnectionInfoRepository.java:65]: As you pointed out, this is the problem. We need to remove client certificate entirely here and elsewhere. Instead we'll use the PEM certificate chain (which includes the client cert at index 0).

**Review by enocom — CHANGES_REQUESTED** [review]

File: alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/DefaultConnectionInfoRepository.java (hunk 1 above)
Comment (ttosta-google) [reply alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/DefaultConnectionInfoRepository.java]: Committed the changes. The integration tests are still failing due to the issues in the `v1` AlloyDB API. 

We should be able to merge this PR in a week when the fix gets to `prod`.

**Comment by ttosta-google** [comment]
I will create another PR once we are ready to test it in production.

**Pull request closed without merging**